dev/http.go

```diff
@@ -147,12 +148,15 @@ func (h *HTTPServer) proxyReq(w http.ResponseWriter, req *http.Request) error {
 	}
 
 	if app.Public && req.URL.Path != "/" {
-		path := filepath.Join(app.dir, "public", path.Clean(req.URL.Path))
+		safeURLPath := path.Clean(req.URL.Path)
+		path := filepath.Join(app.dir, "public", safeURLPath)
 
 		fi, err := os.Stat(path)
 		if err == nil && !fi.IsDir() {
-			http.ServeFile(w, req, path)
-			return httputil.ErrHandled
+			if ofile, err := os.Open(path); err == nil {
+				http.ServeContent(w, req, req.URL.Path, fi.ModTime(), io.ReadSeeker(ofile))
```

Comment:
https://golang.org/pkg/net/http/#ServeFile

> As another special case, `ServeFile` redirects any request where `r.URL.Path` ends in "/index.html" to the same path, without the final "index.html". To avoid such redirects either modify the path or use `ServeContent`.

Reply:
the downside with this approach is that you can't just drop an `index.html` in public and go to `mysite.localhost`. you have to navigate to `mysite.localhost/index.html`.

but, `puma-dev` never supported that anyway, given the `req.URL.Path != "/"`

I figure this is a good enhancement to consider when tackling #87

Reply:
I think this is fine, your justification makes sense.